File: 알고리즘 스터디 문제들/Part 6 - DFS와 BFS/문제/바이러스.cpp
#include <iostream>
#include <array>
#include <vector>
using namespace std;

int cnt = 0;
array<vector<int>, 101> graph;
array<bool, 101> visited = {};

void dfs(int cur)
{
    visited[cur] = true;
    for (size_t i = 0; i < graph[cur].size(); i++)
    {
        if (!visited[graph[cur][i]])
        {
            dfs(graph[cur][i]);
            cnt++;
        }
    }
}

int main()
{
    int N, M, node1, node2;
    cin >> N >> M;

    for (size_t i = 0; i < M; i++)
    {
        cin >> node1 >> node2;
        graph[node1].push_back(node2);
        graph[node2].push_back(node1);
    }

    dfs(1);
    cout << cnt;
}

File: 알고리즘 스터디 문제들/Part 6 - DFS와 BFS/문제/바이러스.py
cnt = 0


def dfs(cur):
    global cnt
    visited[cur] = True
    for node in graph[cur]:
        if (not visited[node]):
            dfs(node)
            cnt += 1


graph = [[] for i in range(101)]
visited = [False for i in range(101)]

N = int(input())
M = int(input())
for i in range(M):
    node1, node2 = map(int, input().split())
    graph[node1].append(node2)
    graph[node2].append(node1)

dfs(1)
print(cnt)
